[#5178] More doc fixes.

dumol opened the pull request:
As per the associated Trac ticket, added an example for using a dedicated volume for user data.

**Drive-by fixes**:
  * unified back the per-distro setup scripts, as documented in README
  * moved customization section to the end
  * replaced shorthand commands for `docker` with their full-blown counterparts
  * syntax fixes.

Tested all three OS'es with current trial version.

dumol committed:
Improved README, also mentioning Alpine.

dumol committed:
Added mention for using volumes for user data.

dumol committed:
Unified install scripts back into the documented sftpplus-docker-setup.sh file.

dumol committed:
Fixes for unified sftpplus-docker-setup.sh script.

dumol committed:
Cosmetic fixes.

adiroiban commented on README.rst:
```diff
@@ -27,6 +27,12 @@ the default `Dockerfile` presented here is not suitable for production.
+mounted from the Docker host.
+This will allow the data to persist when the container no longer exists,
+and also ease access to the data outside of the container.
+For production usage, dedicated volumes should be used for user data.
```
Instead of this info, I was expecting that the README will contain an example of creating a volume and then start the container with /srv/volume mapped to the dedicated volume.

adiroiban commented on sftpplus-docker-setup.sh:
```diff
@@ -0,0 +1,58 @@
+# To be used with the SFTPPlus Dockerfile.
```
this is ok... and is on my taste.... but for the purpose of the example and learning not sure if it is not better to keep thing simple and have separate files for each OK.... I don't know  :)

dumol replied:
It think things are more simple with one common script. Documentation is simpler and the structure of the repo is simplified. True, at the cost of a slightly more complex setup script. But even this comes with the benefit of highlighting OS-dependent commands with the `case` constructs.

adiroiban reviewed (changes requested):
No hard opinion on the unified script :)... but please update the example to demonstrate creating a dedicated volume and then starting the container to use that volume.

I am ok if instead of the dedicated volume we use a mount folder on the docker host... the important is to show how to persist the data.

--------

I know that most people using Docker should already be aware of volume ... but I guess that those people will aready know how to "dockerize" an app and then will not  end up on this repo anyway.

So, I should provide a bit of extra help for the users of this repo and try to create a fully functional example, as close as possible to what they will be using in production.

Thanks!

dumol committed:
Improved comments for the docker setup script.

dumol committed:
Replaced shorthand forms in Docker commands.

dumol committed:
Moved customization section to the end of README.

dumol committed:
Added instructions for using a dedicated volume for storage.

dumol committed:
Syntax fixes.

dumol commented:
Thanks for the review!

To keep things simple, I've added a separate example for creating a dedicated volume and then starting the container to use that volume. All this in the _Image Customization_ section, which I've moved to the end, as it seemed more logical.

dumol commented:
Not sure what is the GitHub workflow when you have a review requesting changes… You should dismiss the review before reassigning to someone else?

adiroiban commented on README.rst:
```diff
@@ -95,49 +65,133 @@ Launching a container
+
+For production usage you should add your own SSH keys and SSL certificates to
+the ``configuration/`` subdirectory.
+Then modify ``server.ini`` to use these files.
```
This is ok.
Not  sure if we should mention that they can also include the content of the certs inside the server.ini

dumol replied:
Sure, no problem.

adiroiban reviewed (approved):
Great work. Thanks! 

-------
 
no need to dismiss the previous review... just assing

adiroiban commented:
You can merge this when you are happy with the PRs content :) I am happy with it :)

dumol committed:
Minor improvements.
